tests/test_main.py

```
@@ -46,3 +46,44 @@ def test_import_without_execution(self) -> None:
+def test_main_validates_http_config(monkeypatch):
+    """Test main validates HTTP configuration."""
+    monkeypatch.setenv("MCP_TRANSPORT", "http")
+    # Don't set port - should fail validation
+    monkeypatch.delenv("MCP_PORT", raising=False)
+
+    try:
+        main()
+        msg = "Should have raised ValueError"
+        raise AssertionError(msg)
+    except ValueError as e:
+        assert "HTTP transport requires MCP_PORT" in str(e)
```

Ruff (0.14.4)

Found assertion on exception `e` in `except` block, use `pytest.raises()` instead

(PT017)